app/assets/javascripts/components/ReportFilter.jsx

```diff
@@ -55,65 +72,82 @@ class ReportFilter extends React.Component {
-      step: 1,
-      orientation: 'horizontal', // 'horizontal' or 'vertical',
-      behaviour: 'tap-drag',
-      range: {
-        'min': 0,
-        'max': this.props.highest_tax_counts.highest_nt_zscore
-      },
-      format: wNumb({
-        decimals: 0
-      })
-    });
-    noUiSlider.create(nt_rpm, {
-      start: [this.state.nt_rpm_start, this.state.nt_rpm_end],
-      connect: true,
-      step: 1,
-      orientation: 'horizontal', // 'horizontal' or 'vertical'
-      range: {
-        'min': 0,
-        'max': this.props.highest_tax_counts.highest_nt_rpm
-      },
-      format: wNumb({
-        decimals: 0
-      })
-    });
-    nt_zscore.noUiSlider.on('update', (values, handle) => {
-      if (handle === 0) {
-        this.setState({
-          nt_zscore_start: Math.round(values[handle])
-        });
-      } else {
-        this.setState({
-          nt_zscore_end: Math.round(values[handle])
-        });
-      }
-    });
-    nt_rpm.noUiSlider.on('update', (values, handle) => {
-      if (handle === 0) {
-        this.setState({
-          nt_rpm_start: Math.round(values[handle])
-        });
-      } else {
-        this.setState({
-          nt_rpm_end: Math.round(values[handle])
-        });
-      }
-    });
+
+    if (this.has_valid_nt_zscore_range) {
+      const nt_zscore = document.getElementById('nt_zscore');
+      noUiSlider.create(nt_zscore, {
+        start: [this.state.nt_zscore_start, this.state.nt_zscore_end],
+        connect: true,
+        step: 1,
+        orientation: 'horizontal', // 'horizontal' or 'vertical',
+        behaviour: 'tap-drag',
+        range: {
+          min: this.props.highest_tax_counts.lowest_nt_zscore,
+          max: this.props.highest_tax_counts.highest_nt_zscore
+        },
+        format: wNumb({
+          decimals: 0
+        })
+      });
+
+      nt_zscore.noUiSlider.on('update', (values, handle) => {
+        if (handle === 0) {
+          this.setState({
+            nt_zscore_start: Math.round(values[handle])
+          });
+        } else {
+          this.setState({
+            nt_zscore_end: Math.round(values[handle])
+          });
+        }
+      });
+    }
+
+    if (this.has_valid_nt_rpm_range) {
+      const nt_rpm = document.getElementById('nt_rpm');
+      noUiSlider.create(nt_rpm, {
+        start: [this.state.nt_rpm_start, this.state.nt_rpm_end],
+        connect: true,
+        step: 1,
+        orientation: 'horizontal', // 'horizontal' or 'vertical'
+        range: {
+          min: 0,
+          max: this.props.highest_tax_counts.highest_nt_rpm
+        },
+        format: wNumb({
+          decimals: 0
+        })
+      });
+      nt_rpm.noUiSlider.on('update', (values, handle) => {
+        if (handle === 0) {
+          this.setState({
+            nt_rpm_start: Math.round(values[handle])
+          });
+        } else {
+          this.setState({
+            nt_rpm_end: Math.round(values[handle])
+          });
+        }
+      });
+    }
```

Comment:
Is it typical to duplicate code like that in JS?   For example, when you figure out the correct rounding math, if the code is duplicated, you will need to fix in two places.